src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        if (password == null || password.isEmpty()) {
+            throw new AuthenticationException("Password can't be empty");
+        }
+        User user = userService.findByEmail(email).orElseThrow(() ->
+                new AuthenticationException("User with email " + email + " not found"));
+        if (user.checkPassword(password)) {
+            return user;
+        } else {
+            throw new AuthenticationException("Wrong Password");
+        }
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {

```

Recommendation: validate `email` (null/empty) before calling `userService.findByEmail(email)`. Passing a null email through to the DAO may produce unclear behavior; fail fast with a clear RegistrationException/AuthenticationException for invalid input.